【bug】npm run build error

The problem is as follows:

dry@ubuntu:~/src/jupyterlab$ npm run build

> @ build /home/dry/src/jupyterlab
> npm run build:packages && cd jupyterlab && npm run build


> @ build:packages /home/dry/src/jupyterlab
> cd packages/all-packages && npm run build


> @jupyterlab/all-packages@0.7.1 build /home/dry/src/jupyterlab/packages/all-packages
> tsc && node build.js

../console-extension/src/index.ts(264,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
  Index signatures are incompatible.
    Type 'ReadonlyJSONValue' is not assignable to type 'JSONValue'.
      Type 'ReadonlyJSONObject' is not assignable to type 'JSONValue'.
        Type 'ReadonlyJSONObject' is not assignable to type 'JSONArray'.
          Property 'length' is missing in type 'ReadonlyJSONObject'.
../console-extension/src/index.ts(278,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../console-extension/src/index.ts(292,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../console-extension/src/index.ts(306,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../console-extension/src/index.ts(320,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../console-extension/src/index.ts(337,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../console-extension/src/index.ts(351,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../console-extension/src/index.ts(393,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(508,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(520,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(531,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(544,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(555,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(566,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(590,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(606,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(635,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(648,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(661,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(672,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(683,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(697,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(708,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(719,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(730,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(741,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(752,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(763,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(774,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(785,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(796,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(807,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(818,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(829,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(840,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(851,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(862,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(873,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(884,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(895,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(906,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(917,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(928,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(939,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(950,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(961,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(969,11): error TS2322: Type '{ path: string; preferredLanguage: string; activate: ReadonlyJSONValue; }' is not assignable to type 'JSONObject'.
../notebook-extension/src/index.ts(969,11): error TS2322: Type '{ path: string; preferredLanguage: string; activate: ReadonlyJSONValue; }' is not assignable to type 'JSONObject'.
  Property 'activate' is incompatible with index signature.
    Type 'ReadonlyJSONValue' is not assignable to type 'JSONValue'.
      Type 'ReadonlyJSONObject' is not assignable to type 'JSONValue'.
        Type 'ReadonlyJSONObject' is not assignable to type 'JSONArray'.
../notebook-extension/src/index.ts(981,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(992,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1003,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1014,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1025,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1036,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1047,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1058,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1069,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1080,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1091,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1102,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1113,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.
../notebook-extension/src/index.ts(1124,32): error TS2345: Argument of type 'ReadonlyJSONObject' is not assignable to parameter of type 'JSONObject'.

--- a/packages/console-extension/src/index.ts
+++ b/packages/console-extension/src/index.ts
@@ -35,7 +35,7 @@ import {
 } from '@phosphor/algorithm';
 
 import {
-  JSONObject
+  ReadonlyJSONObject
 } from '@phosphor/coreutils';
 
 import {
@@ -248,7 +248,7 @@ function activateConsole(app: JupyterLab, manager: IServiceManager, rendermime: 
   palette.addItem({ command, category });
 
   // Get the current widget and activate unless the args specify otherwise.
-  function getCurrent(args: JSONObject): ConsolePanel | null {
+  function getCurrent(args: ReadonlyJSONObject): ConsolePanel | null {
     let widget = tracker.currentWidget;
     let activate = args['activate'] !== false;
     if (activate && widget) {
@@ -371,7 +371,7 @@ function activateConsole(app: JupyterLab, manager: IServiceManager, rendermime: 
 
   command = CommandIDs.inject;
   commands.addCommand(command, {
-    execute: (args: JSONObject) => {
+    execute: args => {
       let path = args['path'];
       tracker.find(widget => {
         if (widget.console.session.path === path) {

--- a/packages/notebook-extension/src/index.ts
+++ b/packages/notebook-extension/src/index.ts
@@ -39,7 +39,7 @@ import {
 } from '@jupyterlab/services';
 
 import {
-  JSONObject
+  ReadonlyJSONObject
 } from '@phosphor/coreutils';
 
 import {
@@ -486,7 +486,7 @@ function addCommands(app: JupyterLab, services: IServiceManager, tracker: Notebo
   const { commands, shell } = app;
 
   // Get the current widget and activate unless the args specify otherwise.
-  function getCurrent(args: JSONObject): NotebookPanel | null {
+  function getCurrent(args: ReadonlyJSONObject): NotebookPanel | null {
     let widget = tracker.currentWidget;
     let activate = args['activate'] !== false;
     if (activate && widget) {
@@ -966,7 +966,7 @@ function addCommands(app: JupyterLab, services: IServiceManager, tracker: Notebo
       if (!widget) {
         return;
       }
-      let options: JSONObject = {
+      let options: ReadonlyJSONObject = {
         path: widget.context.path,
         preferredLanguage: widget.context.model.defaultKernelLanguage,
         activate: args['activate']

